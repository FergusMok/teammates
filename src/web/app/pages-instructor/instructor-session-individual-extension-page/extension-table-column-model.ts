export interface StudentExtensionTableColumnModel {
    sectionName: string;
    teamName: string;
    name: string;
    email: string;
    extensionDeadline: number;
    hasExtension: boolean;
    selected: boolean;
  }

  export interface InstructorExtensionTableColumnModel {
    institute?: string;
    name: string;
    email: string;
    extensionDeadline: number;
    hasExtension: boolean;
    selected: boolean;
  }
