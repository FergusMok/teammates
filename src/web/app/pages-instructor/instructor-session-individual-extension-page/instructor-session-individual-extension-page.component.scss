.background-color-medium-gray {
  background-color: #E5E5E6;
}

.remind-all {
  width: 250px;
}

.clickable {
  cursor: pointer;
}

.margin-right-5px {
  margin-right: 5px;
}

.card-plain {
  background-color: #EAEFF5;
  background-image: none;
  border: 0;
  box-shadow: 0 0 0 0;
}

.close-header {
  padding-top: .75rem;
  padding-right: 1.25rem;
  margin-bottom: 0;
}

.font-bold {
  font-weight: bold;
}

.border-left-gray {
  border-left: 1px solid #DDD;
}

.margin-top-20px {
  margin-top: 20px;
}

.session-form-buttons > button {
  margin-left: var(--btn-margin);
}

.align-row {
  display: flex;
  flex-direction: row;
  gap: 20px;
}
