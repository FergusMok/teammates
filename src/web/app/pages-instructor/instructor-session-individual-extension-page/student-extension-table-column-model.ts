export interface StudentExtensionTableColumnModel {
    sectionName: string;
    teamName: string;
    studentName: string;
    studentEmail: string;
    studentExtensionDeadline: number;
    hasExtension: boolean;
    selected: boolean;
  }
