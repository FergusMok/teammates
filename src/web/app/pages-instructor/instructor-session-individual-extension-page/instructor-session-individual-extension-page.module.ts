import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule, Routes } from '@angular/router';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { TeammatesCommonModule } from '../../components/teammates-common/teammates-common.module';
import { IndividualExtensionConfirmModalComponent } from './individual-extension-confirm-modal/individual-extension-confirm-modal.component';
import { IndividualExtensionDateModalComponent } from './individual-extension-date-modal/individual-extension-date-modal.component';
import { InstructorSessionIndividualExtensionPageComponent } from './instructor-session-individual-extension-page.component';
import { TimepickerModule } from '../../components/timepicker/timepicker.module';
import { DatepickerModule } from '../../components/datepicker/datepicker.module';

const routes: Routes = [
    {
      path: '',
      component: InstructorSessionIndividualExtensionPageComponent,
    },
  ];  

@NgModule({
    imports: [
      CommonModule,
      ReactiveFormsModule,
      TeammatesCommonModule,  
      FormsModule,
      RouterModule.forChild(routes),
      LoadingSpinnerModule,
      DatepickerModule,
      TimepickerModule,
    ],
    entryComponents: [
      IndividualExtensionDateModalComponent,
      IndividualExtensionConfirmModalComponent
    ],
    declarations: [
      InstructorSessionIndividualExtensionPageComponent,
      IndividualExtensionDateModalComponent,
      IndividualExtensionConfirmModalComponent,
    ],
    exports: [
        InstructorSessionIndividualExtensionPageComponent
    ],
  })
  
export class InstructorSessionIndividualExtensionPageModule { }