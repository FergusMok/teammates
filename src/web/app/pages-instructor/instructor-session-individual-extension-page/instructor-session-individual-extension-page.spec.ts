import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { ActivatedRoute, RouterModule } from '@angular/router';
import { of, throwError } from 'rxjs';
import { CourseService } from '../../../services/course.service';
import { FeedbackSessionsService } from '../../../services/feedback-sessions.service';
import { StudentService } from '../../../services/student.service';
import { TimezoneService } from '../../../services/timezone.service';
import {
  Course, FeedbackSession, FeedbackSessionPublishStatus, FeedbackSessionSubmissionStatus, Student,
  Students,
} from '../../../types/api-output';
import { ResponseVisibleSetting, SessionVisibleSetting } from '../../../types/api-request';
import { InstructorSessionIndividualExtensionPageComponent }
  from './instructor-session-individual-extension-page.component';
import { InstructorSessionIndividualExtensionPageModule } from './instructor-session-individual-extension-page.module';

describe('InstructorSessionIndividualExtensionPageComponent', () => {
    const testCourse: Course = {
      courseId: 'exampleId',
      courseName: 'Example Course',
      institute: 'Test Institute',
      timeZone: 'UTC',
      creationTimestamp: 0,
      deletionTimestamp: 1000,
    };

    const testFeedbackSession: FeedbackSession = {
        courseId: 'testId1',
        timeZone: 'UTC',
        feedbackSessionName: 'Test Session',
        instructions: 'Instructions',
        submissionStartTimestamp: 1000000000000,
        submissionEndTimestamp: 1500000000000,
        gracePeriod: 0,
        sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
        responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
        submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
        publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
        isClosingEmailEnabled: true,
        isPublishedEmailEnabled: true,
        createdAtTimestamp: 0,
    };

    const testStudent1: Student = {
      email: 'alice@tmms.com',
      courseId: 'testId',
      name: 'Alice',
      teamName: 'Team 1',
      sectionName: 'Section 1',
    };
    const testStudent2: Student = {
      email: 'bob@tmms.com',
      courseId: 'testId',
      name: 'Bob',
      teamName: 'Team 1',
      sectionName: 'Section 1',
    };
    const testStudent3: Student = {
      email: 'alex@tmms.com',
      courseId: 'testId',
      name: 'alex',
      teamName: 'Team 1',
      sectionName: 'Section 1',
    };
    const students: Students = {
      students: [testStudent1, testStudent2, testStudent3],
    };

    const testTimeString = '5 Apr 2000 2:00:00';

    let component: InstructorSessionIndividualExtensionPageComponent;
    let fixture: ComponentFixture<InstructorSessionIndividualExtensionPageComponent>;
    let studentService: StudentService;
    let courseService: CourseService;
    let feedbackSessionsService: FeedbackSessionsService;
    let timezoneService: TimezoneService;

    beforeEach(waitForAsync(() => {
        TestBed.configureTestingModule({
          imports: [
            HttpClientTestingModule,
            RouterModule,
            InstructorSessionIndividualExtensionPageModule,
          ],
          providers: [StudentService, CourseService, TimezoneService, FeedbackSessionsService,
            {
              provide: ActivatedRoute,
              useValue: {
                queryParams: of({
                  courseid: testCourse.courseId,
                  feedbackSessionName: testFeedbackSession.feedbackSessionName,
                }),
              },
            },
          ],
        })
        .compileComponents();
      }));

    beforeEach(() => {
        fixture = TestBed.createComponent(InstructorSessionIndividualExtensionPageComponent);
        component = fixture.componentInstance;
        studentService = TestBed.inject(StudentService);
        courseService = TestBed.inject(CourseService);
        feedbackSessionsService = TestBed.inject(FeedbackSessionsService);
        timezoneService = TestBed.inject(TimezoneService);
        jest.spyOn(timezoneService, 'guessTimezone').mockReturnValue('UTC');
        fixture.detectChanges();
      });

    it('should create', () => {
        expect(component).toBeTruthy();
      });

    it('should snap with default fields', () => {
        expect(component).toBeTruthy();
    });

    it('should snap with student session loading', () => {
      jest.spyOn(timezoneService, 'formatToString').mockReturnValue(testTimeString);
      component.isLoadingAllStudents = true;
      component.isLoadingFeedbackSession = false;
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });

    it('should snap when feedback session loading', () => {
      jest.spyOn(timezoneService, 'formatToString').mockReturnValue(testTimeString);
      component.isLoadingAllStudents = false;
      component.isLoadingFeedbackSession = true;
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });

    it('should stop loading if student service returns 404', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(throwError({
        status: 404,
        error: { message: 'This is a test message' },
      }));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));

      component.ngOnInit();

      expect(component.isLoadingAllStudents).toBeFalsy();
      expect(component.hasLoadedAllStudentsFailed).toBeTruthy();
      expect(component.isLoadingFeedbackSession).toBeFalsy();
      expect(component.hasLoadingFeedbackSessionFailed).toBeFalsy();
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });

    it('should stop loading if feedback session service returns 404', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(throwError({
        status: 404,
        error: { message: 'This is a test message' },
      }));
      jest.spyOn(timezoneService, 'formatToString').mockReturnValue(testTimeString);

      component.ngOnInit();

      expect(component.isLoadingAllStudents).toBeFalsy();
      expect(component.hasLoadedAllStudentsFailed).toBeFalsy();
      expect(component.isLoadingFeedbackSession).toBeFalsy();
      expect(component.hasLoadingFeedbackSessionFailed).toBeTruthy();
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });

    it('should stop loading if course service returns 404', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(throwError({
        status: 404,
        error: { message: 'This is a test message' },
      }));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));
      jest.spyOn(timezoneService, 'formatToString').mockReturnValue(testTimeString);

      component.ngOnInit();

      expect(component.isLoadingAllStudents).toBeFalsy();
      expect(component.hasLoadedAllStudentsFailed).toBeFalsy();
      expect(component.isLoadingFeedbackSession).toBeFalsy();
      expect(component.hasLoadingFeedbackSessionFailed).toBeTruthy();
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });

    it('should snap with details and extended students', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));

      component.ngOnInit();

      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });

    it('should snap when clicking the Select All button', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));

      component.ngOnInit();
      fixture.detectChanges();
      const selectAllButton = fixture.debugElement.query(By.css('#select-all-btn'));
      selectAllButton.triggerEventHandler('click', null);

      expect(fixture).toMatchSnapshot();
    });

    it('should disable extend and delete button when no student selected', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));
      component.ngOnInit();
      fixture.detectChanges();

      const extendButton: any = fixture.debugElement.nativeElement.querySelector('#extend-btn');
      const deleteButton: any = fixture.debugElement.nativeElement.querySelector('#delete-btn');

      expect(extendButton.textContent).toEqual('Extend');
      expect(extendButton.disabled).toBeTruthy();
      expect(deleteButton.textContent).toEqual('Delete');
      expect(deleteButton.disabled).toBeTruthy();
    });

    it('should enable the extend button when a student is selected', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));
      component.ngOnInit();
      component.studentsOfCourse[0].selected = true;
      fixture.detectChanges();

      const extendButton: any = fixture.debugElement.nativeElement.querySelector('#extend-btn');
      const deleteButton: any = fixture.debugElement.nativeElement.querySelector('#delete-btn');

      expect(extendButton.textContent).toEqual('Extend');
      expect(extendButton.disabled).toBeFalsy();
      expect(deleteButton.textContent).toEqual('Delete');
      expect(deleteButton.disabled).toBeTruthy();
    });

    it('should enable extend and delete button when student with extension selected', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));
      component.ngOnInit();
      component.studentsOfCourse[0].hasExtension = true;
      component.studentsOfCourse[0].selected = true;
      fixture.detectChanges();

      const extendButton: any = fixture.debugElement.nativeElement.querySelector('#extend-btn');
      const deleteButton: any = fixture.debugElement.nativeElement.querySelector('#delete-btn');

      expect(extendButton.textContent).toEqual('Extend');
      expect(extendButton.disabled).toBeFalsy();
      expect(deleteButton.textContent).toEqual('Delete');
      expect(deleteButton.disabled).toBeFalsy();
    });

    it('should disable delete button if one of selected students does not have extension', () => {
      jest.spyOn(studentService, 'getStudentsFromCourse').mockReturnValue(of(students));
      jest.spyOn(courseService, 'getCourseAsInstructor').mockReturnValue(of(testCourse));
      jest.spyOn(feedbackSessionsService, 'getFeedbackSession').mockReturnValue(of(testFeedbackSession));
      component.ngOnInit();
      component.studentsOfCourse[0].hasExtension = true;
      component.studentsOfCourse[0].selected = true;
      component.studentsOfCourse[1].selected = true;
      fixture.detectChanges();

      const extendButton: any = fixture.debugElement.nativeElement.querySelector('#extend-btn');
      const deleteButton: any = fixture.debugElement.nativeElement.querySelector('#delete-btn');

      expect(extendButton.textContent).toEqual('Extend');
      expect(extendButton.disabled).toBeFalsy();
      expect(deleteButton.textContent).toEqual('Delete');
      expect(deleteButton.disabled).toBeFalsy();
    });

});
