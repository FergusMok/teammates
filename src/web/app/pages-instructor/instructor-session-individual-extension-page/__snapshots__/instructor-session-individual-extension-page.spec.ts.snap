// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualExtensionDateModalComponent should snap when clicking the Select All button 1`] = `
<tm-instructor-session-individual-extension-page
  DATETIME_FORMAT={[Function String]}
  SortBy={[Function Object]}
  SortOrder={[Function Object]}
  columnsData={[Function Array]}
  courseId={[Function String]}
  courseName={[Function String]}
  courseService={[Function CourseService]}
  feedbackSessionEndingTime={[Function Number]}
  feedbackSessionName="undefined"
  feedbackSessionTimeZone={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasLoadedAllStudentsFailed="false"
  hasLoadingFeedbackSessionFailed="false"
  isAllSelected="false"
  isLoadingAllStudents="false"
  isLoadingFeedbackSession="false"
  isOpen={[Function Boolean]}
  ngbModal={[Function NgbModal]}
  route={[Function Object]}
  rowsData={[Function Array]}
  sortBy={[Function Number]}
  sortOrder="0"
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  studentsOfCourse={[Function Array]}
  tableComparatorService={[Function TableComparatorService]}
  timezoneService={[Function TimezoneService]}
>
  <div>
    <div
      class="card border-primary margin-top-20px"
    >
      <div
        class="card-body"
      >
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Course ID
          </div>
          <div
            class="col-md-3 text-md-left"
            id="time-zone"
          >
             exampleId 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Time Zone
          </div>
          <div
            class="col-md-3 text-md-left"
            id="time-zone"
          >
             UTC 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Course Name
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
             Example Course 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Session Name
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Original Deadline
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
             5 Jul 2017 2:40:00 
          </div>
        </div>
      </div>
    </div>
    <div
      class="container ml-1"
    >
      <div
        class="align-row"
      >
        <div>
          <button
            class="btn btn-success"
            disabled=""
            id="extend-btn"
            type="button"
          >
            Extend
          </button>
        </div>
        <div>
          <button
            class="btn btn-danger"
            id="delete-btn"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
      <small>
        Select students from list below and click apply or remove to change their feedback session deadline extension.
      </small>
    </div>
  </div><div>
    <div
      class="modal-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-bordered"
          id="student-list-table"
        >
          <thead>
            <tr>
              <th>
                <label>
                  <input
                    id="select-all-btn"
                    type="checkbox"
                  />
                   Select all
                </label>
              </th>
              <th
                class="sortable-header"
              >
                 Section 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                  <i
                    class="fas fa-sort-down"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Team 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Name 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Email 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Current Deadline 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
            </tr>
          </thead>
          <tr>
            <td
              class="align-center"
            >
              <label>
                <input
                  type="checkbox"
                />
              </label>
            </td>
            <td>
              Section 1
            </td>
            <td>
              Team 1
            </td>
            <td>
              Alice
            </td>
            <td>
              alice@tmms.com
            </td>
            <td>
              5 Jul 2017 2:40:00
            </td>
          </tr>
          <tr>
            <td
              class="align-center"
            >
              <label>
                <input
                  type="checkbox"
                />
              </label>
            </td>
            <td>
              Section 1
            </td>
            <td>
              Team 1
            </td>
            <td>
              Bob
            </td>
            <td>
              bob@tmms.com
            </td>
            <td>
              5 Jul 2017 2:40:00
            </td>
          </tr>
          <tr>
            <td
              class="align-center"
            >
              <label>
                <input
                  type="checkbox"
                />
              </label>
            </td>
            <td>
              Section 1
            </td>
            <td>
              Team 1
            </td>
            <td>
              alex
            </td>
            <td>
              alex@tmms.com
            </td>
            <td>
              5 Jul 2017 2:40:00
            </td>
          </tr>
        </table>
      </div>
    </div>
  </div>
</tm-instructor-session-individual-extension-page>
`;

exports[`IndividualExtensionDateModalComponent should snap when feedback session loading 1`] = `
<tm-instructor-session-individual-extension-page
  DATETIME_FORMAT={[Function String]}
  SortBy={[Function Object]}
  SortOrder={[Function Object]}
  columnsData={[Function Array]}
  courseId={[Function String]}
  courseName=""
  courseService={[Function CourseService]}
  feedbackSessionEndingTime="0"
  feedbackSessionName="undefined"
  feedbackSessionTimeZone=""
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasLoadedAllStudentsFailed="false"
  hasLoadingFeedbackSessionFailed="false"
  isAllSelected="false"
  isLoadingAllStudents="false"
  isLoadingFeedbackSession={[Function Boolean]}
  isOpen={[Function Boolean]}
  ngbModal={[Function NgbModal]}
  route={[Function Object]}
  rowsData={[Function Array]}
  sortBy={[Function Number]}
  sortOrder="0"
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  studentsOfCourse={[Function Array]}
  tableComparatorService={[Function TableComparatorService]}
  timezoneService={[Function TimezoneService]}
>
  <tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner><div>
    <div
      class="modal-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-bordered"
          id="student-list-table"
        >
          <thead>
            <tr>
              <th>
                <label>
                  <input
                    id="select-all-btn"
                    type="checkbox"
                  />
                   Select all
                </label>
              </th>
              <th
                class="sortable-header"
              >
                 Section 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                  <i
                    class="fas fa-sort-down"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Team 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Name 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Email 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Current Deadline 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </div>
</tm-instructor-session-individual-extension-page>
`;

exports[`IndividualExtensionDateModalComponent should snap with details and extended students 1`] = `
<tm-instructor-session-individual-extension-page
  DATETIME_FORMAT={[Function String]}
  SortBy={[Function Object]}
  SortOrder={[Function Object]}
  columnsData={[Function Array]}
  courseId={[Function String]}
  courseName={[Function String]}
  courseService={[Function CourseService]}
  feedbackSessionEndingTime={[Function Number]}
  feedbackSessionName="undefined"
  feedbackSessionTimeZone={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasLoadedAllStudentsFailed="false"
  hasLoadingFeedbackSessionFailed="false"
  isAllSelected="false"
  isLoadingAllStudents="false"
  isLoadingFeedbackSession="false"
  isOpen={[Function Boolean]}
  ngbModal={[Function NgbModal]}
  route={[Function Object]}
  rowsData={[Function Array]}
  sortBy={[Function Number]}
  sortOrder="0"
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  studentsOfCourse={[Function Array]}
  tableComparatorService={[Function TableComparatorService]}
  timezoneService={[Function TimezoneService]}
>
  <div>
    <div
      class="card border-primary margin-top-20px"
    >
      <div
        class="card-body"
      >
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Course ID
          </div>
          <div
            class="col-md-3 text-md-left"
            id="time-zone"
          >
             exampleId 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Time Zone
          </div>
          <div
            class="col-md-3 text-md-left"
            id="time-zone"
          >
             UTC 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Course Name
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
             Example Course 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Session Name
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Original Deadline
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
             5 Jul 2017 2:40:00 
          </div>
        </div>
      </div>
    </div>
    <div
      class="container ml-1"
    >
      <div
        class="align-row"
      >
        <div>
          <button
            class="btn btn-success"
            disabled=""
            id="extend-btn"
            type="button"
          >
            Extend
          </button>
        </div>
        <div>
          <button
            class="btn btn-danger"
            id="delete-btn"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
      <small>
        Select students from list below and click apply or remove to change their feedback session deadline extension.
      </small>
    </div>
  </div><div>
    <div
      class="modal-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-bordered"
          id="student-list-table"
        >
          <thead>
            <tr>
              <th>
                <label>
                  <input
                    id="select-all-btn"
                    type="checkbox"
                  />
                   Select all
                </label>
              </th>
              <th
                class="sortable-header"
              >
                 Section 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                  <i
                    class="fas fa-sort-down"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Team 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Name 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Email 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Current Deadline 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
            </tr>
          </thead>
          <tr>
            <td
              class="align-center"
            >
              <label>
                <input
                  type="checkbox"
                />
              </label>
            </td>
            <td>
              Section 1
            </td>
            <td>
              Team 1
            </td>
            <td>
              Alice
            </td>
            <td>
              alice@tmms.com
            </td>
            <td>
              5 Jul 2017 2:40:00
            </td>
          </tr>
          <tr>
            <td
              class="align-center"
            >
              <label>
                <input
                  type="checkbox"
                />
              </label>
            </td>
            <td>
              Section 1
            </td>
            <td>
              Team 1
            </td>
            <td>
              Bob
            </td>
            <td>
              bob@tmms.com
            </td>
            <td>
              5 Jul 2017 2:40:00
            </td>
          </tr>
          <tr>
            <td
              class="align-center"
            >
              <label>
                <input
                  type="checkbox"
                />
              </label>
            </td>
            <td>
              Section 1
            </td>
            <td>
              Team 1
            </td>
            <td>
              alex
            </td>
            <td>
              alex@tmms.com
            </td>
            <td>
              5 Jul 2017 2:40:00
            </td>
          </tr>
        </table>
      </div>
    </div>
  </div>
</tm-instructor-session-individual-extension-page>
`;

exports[`IndividualExtensionDateModalComponent should stop loading if student service returns 404 1`] = `
<tm-instructor-session-individual-extension-page
  DATETIME_FORMAT={[Function String]}
  SortBy={[Function Object]}
  SortOrder={[Function Object]}
  columnsData={[Function Array]}
  courseId={[Function String]}
  courseName={[Function String]}
  courseService={[Function CourseService]}
  feedbackSessionEndingTime={[Function Number]}
  feedbackSessionName="undefined"
  feedbackSessionTimeZone={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasLoadedAllStudentsFailed={[Function Boolean]}
  hasLoadingFeedbackSessionFailed="false"
  isAllSelected="false"
  isLoadingAllStudents="false"
  isLoadingFeedbackSession="false"
  isOpen={[Function Boolean]}
  ngbModal={[Function NgbModal]}
  route={[Function Object]}
  rowsData={[Function Array]}
  sortBy={[Function Number]}
  sortOrder="0"
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  studentsOfCourse={[Function Array]}
  tableComparatorService={[Function TableComparatorService]}
  timezoneService={[Function TimezoneService]}
>
  <div>
    <div
      class="card border-primary margin-top-20px"
    >
      <div
        class="card-body"
      >
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Course ID
          </div>
          <div
            class="col-md-3 text-md-left"
            id="time-zone"
          >
             exampleId 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Time Zone
          </div>
          <div
            class="col-md-3 text-md-left"
            id="time-zone"
          >
             UTC 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Course Name
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
             Example Course 
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Session Name
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
          </div>
        </div>
        <div
          class="row text-center"
        >
          <div
            class="col-md-2 text-md-right font-bold"
          >
            Original Deadline
          </div>
          <div
            class="col-md-10 text-md-left"
            id="course-name"
          >
             5 Jul 2017 2:40:00 
          </div>
        </div>
      </div>
    </div>
    <div
      class="container ml-1"
    >
      <div
        class="align-row"
      >
        <div>
          <button
            class="btn btn-success"
            disabled=""
            id="extend-btn"
            type="button"
          >
            Extend
          </button>
        </div>
        <div>
          <button
            class="btn btn-danger"
            id="delete-btn"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
      <small>
        Select students from list below and click apply or remove to change their feedback session deadline extension.
      </small>
    </div>
  </div><div>
    <div
      class="modal-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-bordered"
          id="student-list-table"
        >
          <thead>
            <tr>
              <th>
                <label>
                  <input
                    id="select-all-btn"
                    type="checkbox"
                  />
                   Select all
                </label>
              </th>
              <th
                class="sortable-header"
              >
                 Section 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                  <i
                    class="fas fa-sort-down"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Team 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Name 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Email 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
              <th
                class="sortable-header"
              >
                 Current Deadline 
                <span
                  class="fa-stack"
                >
                  <i
                    class="fas fa-sort"
                  />
                </span>
              </th>
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </div>
</tm-instructor-session-individual-extension-page>
`;
